src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7172,34 +7173,40 @@ private bool IsEndOfFixedStatement()
                 || this.CurrentToken.Kind == SyntaxKind.SemicolonToken;
         }
 
+        // ParseEmbeddedStatement is called through many recursive statement parsing cases. We want
+        // to ensure it is inlined into the callers as the overhead of this single method can have a
+        // deep impact on the number of recursive calls we can make (more than a hundred during
+        // empirical testing).
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Would we get inlining without this attribute if we refactor the code to the following?
```C#
private StatementSyntax ParseEmbeddedStatement()
{
    return ParseEmbeddedStatementContinued(ParseStatementCore());
}

private StatementSyntax ParseEmbeddedStatementContinued(StatementSyntax statement)
{
    if (statement == null)
    {
        return SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken));
    }
    if (statement.Kind == SyntaxKind.ExpressionStatement)
    {
        // ...
    }
    return statement;
}
```

Reply:
I'll check!

Reply:
yup.  that works.

Reply:
Removed the attribute, and refactored to be simple enough to inline well.